File: analyzer/src/main/scala/io/github/tassiLuca/analyzer/client/view/AnalyzerGUI.scala
package io.github.tassiLuca.analyzer.client.view

import io.github.tassiLuca.analyzer.client.view.ui.MainFrame
import io.github.tassiLuca.analyzer.client.{AnalyzerView, AppController, OrganizationReport}

import javax.swing.SwingUtilities

class AnalyzerGUI(controller: AppController) extends AnalyzerView:
  private val gui: MainFrame = MainFrame(controller)

  override def run(): Unit =
    gui.pack()
    gui.setVisible(true)

  override def update(result: OrganizationReport): Unit =
    SwingUtilities.invokeLater(() => gui.updateResults(result))

  override def error(errorMessage: String): Unit = gui.showError(errorMessage)

  override def endComputation(): Unit = gui.endSession()

File: analyzer-direct/src/main/kotlin/io/github/tassiLuca/analyzerkt/client/AnalyzerGUI.kt
package io.github.tassiLuca.analyzerkt.client

import io.github.tassiLuca.analyzer.commons.client.AppController
import io.github.tassiLuca.analyzer.commons.client.ui.MainFrame
import javax.swing.SwingUtilities

/** The analyzer application GUI. */
class AnalyzerGUI(controller: AppController) : AnalyzerView {
    private val gui: MainFrame = MainFrame(controller)

    override fun run() {
        gui.pack()
        gui.isVisible = true
    }

    override fun update(result: OrganizationReport) = SwingUtilities.invokeLater {
        gui.contributionsModel().setDataVector(
            result.first.toList().sortedByDescending { it.second }.map { arrayOf(it.first, it.second) }.toTypedArray(),
            gui.contributionsCols(),
        )
        gui.repoDetailsModel().setDataVector(
            result.second.map { arrayOf(it.name, it.issues, it.stars, it.lastRelease) }.toTypedArray(),
            gui.repoDetailsCols(),
        )
    }

    override fun endComputation() = SwingUtilities.invokeLater { gui.ended() }

    override fun error(errorMessage: String) = SwingUtilities.invokeLater { gui.showError(errorMessage) }

    override fun cancelled() = SwingUtilities.invokeLater { gui.cancelled() }
}
